macOS build fails

```
#23 10.54 /ddnet-builder/ddnet-source/src/base/system.cpp:2557:88: error: cannot pass non-trivial object of type 'const std::wstring' (aka 'const basic_string<wchar_t>') to variadic function; expected type from format string was 'char *' [-Wnon-pod-varargs]
#23 10.54  2557 |                 log_error("filesystem", "Failed to rename file '%s' to '%s' for removal (%ld '%s')", wide_filename, wide_filename_temp, error, windows_format_system_message(error).c_str());
#23 10.54       |                                                                 ~~                                   ^~~~~~~~~~~~~
#23 10.54 /ddnet-builder/ddnet-source/src/base/log.h:29:55: note: expanded from macro 'log_error'
#23 10.54    29 | #define log_error(sys, ...) log_log(LEVEL_ERROR, sys, __VA_ARGS__)
#23 10.54       |                                                       ^~~~~~~~~~~
#23 10.54 /ddnet-builder/ddnet-source/src/base/system.cpp:2557:103: error: cannot pass non-trivial object of type 'std::wstring' (aka 'basic_string<wchar_t>') to variadic function; expected type from format string was 'char *' [-Wnon-pod-varargs]
#23 10.54  2557 |                 log_error("filesystem", "Failed to rename file '%s' to '%s' for removal (%ld '%s')", wide_filename, wide_filename_temp, error, windows_format_system_message(error).c_str());
#23 10.54       |                                                                         ~~                                          ^~~~~~~~~~~~~~~~~~
#23 10.54 /ddnet-builder/ddnet-source/src/base/log.h:29:55: note: expanded from macro 'log_error'
#23 10.54    29 | #define log_error(sys, ...) log_log(LEVEL_ERROR, sys, __VA_ARGS__)
#23 10.54       |                                                       ^~~~~~~~~~~
#23 10.54 /ddnet-builder/ddnet-source/src/base/system.cpp:2569:67: error: cannot pass non-trivial object of type 'std::wstring' (aka 'basic_string<wchar_t>') to variadic function; expected type from format string was 'char *' [-Wnon-pod-varargs]
#23 10.54  2569 |         log_error("filesystem", "Failed to remove file '%s' (%ld '%s')", wide_filename_temp, error, windows_format_system_message(error).c_str());
#23 10.54       |                                                         ~~               ^~~~~~~~~~~~~~~~~~
```

--- a/src/base/system.cpp
+++ b/src/base/system.cpp
@@ -2569,7 +2569,9 @@ int fs_remove(const char *filename)
 	}
 	const std::string filename_temp = windows_wide_to_utf8(wide_filename_temp.c_str()).value_or("(invalid filename)");
 	log_error("filesystem", "Failed to remove file '%s' (%ld '%s')", filename_temp.c_str(), error, windows_format_system_message(error).c_str());
-	return 1;
+	// Success: While the temporary could not be deleted, this is also considered success because the original file does not exist anymore.
+	//          Callers of this function expect that the original file does not exist anymore if and only if the function succeeded.
+	return 0;
 #else
 	if(unlink(filename) == 0 || errno == ENOENT)
 	{

